### PR TITLE
[Reserva] : add business logic to fetch booking dates and small fixes

### PR DESCRIPTION
**Pull request by pdPinho**
### Issue

Closes [Endpoint para ver horários](https://tqsprojectrestaurant.atlassian.net/jira/software/projects/RMO/list?selectedIssue=RMO-65)

### Reason for this change

We want to be capable of checking the schedule for the chosen day to choose a timeslot and book it.

### Description of changes

- Created controller with desired endpoint
- Created service with desired logic
- Created tests
- Created new restaurant class to provide easy manipulation
- Created ENUM for roles

### Description of how you validated changes

Tested with `mvn test` and `docker-compose up`

### Checklist
- [X] I have tested my changes locally.

### Aditional Information
N/A

### CODE REVIEW

File: src/main/java/tqs/project/api/others/Restaurant.java
Comment (WildBunnie) [inline src/main/java/tqs/project/api/others/Restaurant.java:1]: minor nitpick about mixing english and portuguese in the variable names

**Review by WildBunnie — COMMENTED** [review]
The code seems fine, I just don't understand how the booking works. Looking at AvailableSlots I'm guessing there can only be a booking per slot? If so why is there a "quantidadeMesas"?

**Review by fungame2270 — APPROVED** [review]
lgtm

**Review by fungame2270 — APPROVED** [review]
ta melhor

**Pull request merged**